File: services/api-principal/src/main/java/com/guminteligencia/ura_chatbot_ia/infrastructure/mapper/ClienteMapper.java
package com.guminteligencia.ura_chatbot_ia.infrastructure.mapper;

import com.guminteligencia.ura_chatbot_ia.domain.Cliente;
import com.guminteligencia.ura_chatbot_ia.infrastructure.repository.entity.ClienteEntity;

public class ClienteMapper {
    public static Cliente paraDomain(ClienteEntity entity) {
        return Cliente.builder()
                .id(entity.getId())
                .nome(entity.getNome())
                .telefone(entity.getTelefone())
                .atributosQualificacao(entity.getAtributosQualificacao())
                .inativo(entity.isInativo())
                .usuario(UsuarioMapper.paraDomain(entity.getUsuario()))
                .build();
    }

    public static ClienteEntity paraEntity(Cliente domain) {
        return ClienteEntity.builder()
                .id(domain.getId())
                .nome(domain.getNome())
                .telefone(domain.getTelefone())
                .atributosQualificacao(domain.getAtributosQualificacao())
                .inativo(domain.isInativo())
                .usuario(UsuarioMapper.paraEntity(domain.getUsuario()))
                .build();
    }
}

File: services/api-agente/src/infrastructure/mapper/cliente_mapper.py
from src.infrastructure.entity import ClienteEntity
from src.domain.cliente import Cliente
import uuid

class ClienteMapper:

    def paraDomain(self, clienteEntity: ClienteEntity) -> Cliente:
        return Cliente(
            id=str(uuid.UUID(clienteEntity.id_cliente)),
            nome=clienteEntity.nome,
            telefone=clienteEntity.telefone,
            atributos_qualificacao=clienteEntity.atributos_qualificacao,
            inativo=clienteEntity.inativo,
            usuario_id=str(uuid.UUID(clienteEntity.usuario_id))
        )